scripts/backport.py

```diff
@@ -162,9 +162,12 @@ def git_returncode(command):
     f"Will commit as {os.environ['GIT_COMMITTER_NAME']} <{os.environ['GIT_COMMITTER_EMAIL']}>"
 )
 
-# Fetch the sources
-git_check(f"fetch {source_remote}")
-git_check(f"fetch {target_remote}")
+# Fetch the main branch. Apparently the local repo can be shallow in some cases
+# in Github Actions, so specify the depth. --unshallow will complain on normal
+# repositories, this is why we don't use it here.
+git_check(
+    f"fetch --quiet --depth=1000 {source_remote} main:refs/remotes/{source_remote}/main"
```

Comment:
Nit: We have the 1000 in the `git fetch ... --depth=1000 ...`, `git log -1000 ...`, `...` calls. It might be good introducing a constant for this.